<ui version="4.0" >
 <author></author>
 <comment></comment>
 <exportmacro></exportmacro>
 <class>ProcWizardDeadlinesPage</class>
 <widget class="QWidget" name="ProcWizardDeadlinesPage" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>489</width>
    <height>398</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Form</string>
  </property>
  <layout class="QGridLayout" >
   <property name="margin" >
    <number>9</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item row="0" column="0" >
    <widget class="QLabel" name="label_18" >
     <property name="text" >
      <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;/head>&lt;body style=" white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;">&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">&lt;span style=" font-weight:600;">Select stage deadlines.&lt;/span>&lt;/p>&lt;/body>&lt;/html></string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2" >
    <widget class="Line" name="line" >
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="5" column="1" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>20</width>
       <height>91</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="0" colspan="2" >
    <widget class="QLabel" name="label_5" >
     <property name="text" >
      <string>Each Adder procedure consists of a number of stages that require participation from a number of users.  You may assign to each of these stages a duration, after which no more participation is allowed.  Enter the procedure stage durations below.</string>
     </property>
     <property name="wordWrap" >
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="1" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>20</width>
       <height>91</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="0" colspan="2" >
    <layout class="QGridLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item row="1" column="1" >
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QLabel" name="label_9" >
         <property name="text" >
          <string>Hours</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="secsharehoursSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_10" >
         <property name="text" >
          <string>Minutes</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="secshareminutesSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_11" >
         <property name="text" >
          <string>Seconds</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="secsharesecondsSpinBox" />
       </item>
      </layout>
     </item>
     <item row="3" column="0" >
      <widget class="QLabel" name="label_4" >
       <property name="toolTip" >
        <string/>
       </property>
       <property name="whatsThis" >
        <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;/head>&lt;body style=" white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;">&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">In this stage, users are permitted to log on and vote.&lt;/p>&lt;/body>&lt;/html></string>
       </property>
       <property name="text" >
        <string>Ballot &amp;casting stage:</string>
       </property>
       <property name="buddy" >
        <cstring>ballothoursSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="0" >
      <widget class="QLabel" name="label" >
       <property name="whatsThis" >
        <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;/head>&lt;body style=" white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;">&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">In this stage, authorities generate keypairs and submit their public keys.&lt;/p>&lt;/body>&lt;/html></string>
       </property>
       <property name="text" >
        <string>&amp;Public key stage:</string>
       </property>
       <property name="buddy" >
        <cstring>pubkeyhoursSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="0" >
      <widget class="QLabel" name="label_3" >
       <property name="whatsThis" >
        <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;/head>&lt;body style=" white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;">&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">In this stage, authorities retrieve secret share information from the server, and aggregate it to form their secret keys.&lt;/p>&lt;/body>&lt;/html></string>
       </property>
       <property name="text" >
        <string>Secret &amp;key stage:</string>
       </property>
       <property name="buddy" >
        <cstring>seckeyhoursSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1" >
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QLabel" name="label_15" >
         <property name="text" >
          <string>Hours</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="ballothoursSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_16" >
         <property name="text" >
          <string>Minutes</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="ballotminutesSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_17" >
         <property name="text" >
          <string>Seconds</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="ballotsecondsSpinBox" />
       </item>
      </layout>
     </item>
     <item row="2" column="1" >
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QLabel" name="label_12" >
         <property name="text" >
          <string>Hours</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="seckeyhoursSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_13" >
         <property name="text" >
          <string>Minutes</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="seckeyminutesSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_14" >
         <property name="text" >
          <string>Seconds</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="seckeysecondsSpinBox" />
       </item>
      </layout>
     </item>
     <item row="1" column="0" >
      <widget class="QLabel" name="label_2" >
       <property name="whatsThis" >
        <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;/head>&lt;body style=" white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;">&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">In this stage, authorities generate secret share information, and submit it to the server.&lt;/p>&lt;/body>&lt;/html></string>
       </property>
       <property name="text" >
        <string>&amp;Secret share stage:</string>
       </property>
       <property name="buddy" >
        <cstring>secsharehoursSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1" >
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QLabel" name="label_6" >
         <property name="text" >
          <string>Hours</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="pubkeyhoursSpinBox" >
         <property name="maximum" >
          <number>99</number>
         </property>
         <property name="minimum" >
          <number>0</number>
         </property>
         <property name="value" >
          <number>0</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7" >
         <property name="text" >
          <string>Minutes</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="pubkeyminutesSpinBox" />
       </item>
       <item>
        <widget class="QLabel" name="label_8" >
         <property name="text" >
          <string>Seconds</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="pubkeysecondsSpinBox" />
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <pixmapfunction></pixmapfunction>
 <tabstops>
  <tabstop>pubkeyhoursSpinBox</tabstop>
  <tabstop>pubkeyminutesSpinBox</tabstop>
  <tabstop>pubkeysecondsSpinBox</tabstop>
  <tabstop>secsharehoursSpinBox</tabstop>
  <tabstop>secshareminutesSpinBox</tabstop>
  <tabstop>secsharesecondsSpinBox</tabstop>
  <tabstop>seckeyhoursSpinBox</tabstop>
  <tabstop>seckeyminutesSpinBox</tabstop>
  <tabstop>seckeysecondsSpinBox</tabstop>
  <tabstop>ballothoursSpinBox</tabstop>
  <tabstop>ballotminutesSpinBox</tabstop>
  <tabstop>ballotsecondsSpinBox</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
